### DISCUSSION TITLE
Factorials

### DISCUSSION THREAD

**Discussion by ghost** [TodePond/GulfOfMexico#272] [Ideas]
There is no built in features for calculating the factorials of numbers in DreamBerd. Given the use of standard notation, standard factorial notation could easily be added just like in other languages.

To find the factorial of 3:
`
const var factorial = 3!

// Auto append: Dreamberd will auto append exclamation marks for you

// const var factorial = 6

// const var factorial = 6!

// const var factorial = 720

// const var factorial = 720!

// const var factorial = 2.601 × 10^1746

...
`

This is easy and will make finding the factorials of numbers easier.

**Comment by stohrendorf** [comment]
How does this interact with the arbitrary amount of exclamantion marks allowed? Also, this seems to be a duplicate of #152.

**Reply by ghost** [reply]
Rather than focusing on an integration that mitigates the use of the mark with another exclamation mark, using factorials with numbers directly would be better
